adapter/src/main/java/org/apache/calcite/prepare/KuduPrepareImpl.java

```diff
@@ -366,30 +462,23 @@ else if (kuduTable.getSchema().hasColumn(sqlnode.toString())) {
               kuduTable.getPartitionSchema().getHashBucketSchemas().get(0).getNumBuckets());
         }
 
-        // if there is a row timestamp column defined, create a single dummy range
-        // partition for
-        // that column so that we can add new partitions later
-        if (!rangePartitionCols.isEmpty()) {
-          String rowTimestampColumn = rangePartitionCols.get(0);
-          CalciteKuduTable calciteTable = (CalciteKuduTable) kuduSchema.getTable(kuduTable.getName());
-          PartialRow lowerBound = cubeSchema.newPartialRow();
-          PartialRow upperBound = cubeSchema.newPartialRow();
-
-          if (kuduSchema.createDummyPartition) {
-            // Set range partition to EPOCH.MAX - minvalue for DESC case.
-            if (calciteTable.isColumnOrderedDesc(rowTimestampColumn)) {
-              lowerBound.addTimestamp(rowTimestampColumn,
-                  new Timestamp(CalciteKuduTable.EPOCH_FOR_REVERSE_SORT_IN_MILLISECONDS - Long.MIN_VALUE));
-              upperBound.addTimestamp(rowTimestampColumn,
-                  new Timestamp(CalciteKuduTable.EPOCH_FOR_REVERSE_SORT_IN_MILLISECONDS - Long.MIN_VALUE + 1));
-            } else {
-              lowerBound.addTimestamp(rowTimestampColumn, new Timestamp(Long.MIN_VALUE));
-              upperBound.addTimestamp(rowTimestampColumn, new Timestamp(Long.MIN_VALUE + 1));
-            }
+        // create a single dummy range partition using the rollup timestamp column
+        PartialRow lowerBound = cubeSchema.newPartialRow();
+        PartialRow upperBound = cubeSchema.newPartialRow();
+        if (kuduSchema.createDummyPartition) {
+          // Set range partition to EPOCH.MAX - minvalue for DESC case.
+          if (descendingCols.contains(rollupTimestampColumn)) {
```

Comment:
The range partitioning of the cube is determined by the timestamp column on which the roll up is being done. As long as the base table and cube tables have stores timestamps that are close then range partitioning on different columns should be ok. If there is a large difference in time between the timestamp column of the fact and cube there could be data inconsistency issues while dropping range partitions based on time.